### ISSUE TITLE
Pylint does not catch simple case of UnboundLocalError (variable-undefined/used-before-assignment)

### ISSUE THREAD

**Issue by milosivanovic** [pylint-dev/pylint#9373]
**Pylint does not catch a simple case of variable-undefined and/or used-before-assignment:**

### Bug description

```python
def func():
    if False:
        chk = "something"
    else:
        print(chk)

func()
```

When executed, the `else` block in the above example would raise `UnboundLocalError: cannot access local variable 'chk' where it is not associated with a value`

Is this a regression of https://github.com/pylint-dev/pylint/issues/4045 and https://github.com/pylint-dev/pylint/issues/5811?

### Command used

```shell
pylint --disable=using-constant-test,missing-module-docstring,missing-function-docstring
```

### Pylint output

```python
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

Pylint should easily catch `chk` as `used-before-assignment` and/or `undefined-variable`

### Pylint version

```shell
pylint 3.0.3
astroid 3.0.2
Python 3.11.7 (main, Dec  4 2023, 18:10:11) [Clang 15.0.0 (clang-1500.1.0.2.5)]
```

### OS / Environment

macOS Sonoma 14.2.1 (23C71)

**Comment by jacobtylerwalls** [comment]
Duplicate of #1727

Happy to have your review of the PR to close it!
